horace_core/sqw/@Experiment/private/build_from_old_headers_.m

```diff
@@ -1,37 +1,62 @@
 function obj = build_from_old_headers_(~,varargin)
-% in this case the header (which is what varargin{1} is in
-% this case) is a cell of runs. Consequently we run over
-% the runs in the cell doing just what we did to one run
-% header in the if block above
+%
+% Input
+% ------
+%   >> ~ : the object (an Experiment), not used - a completely new
+%          Experiment is created and output at the end
+%   >> headers == varargin{1}: cell array of old-style header structs
+%
+% Output
+% -------
+%   >> obj: an Experiment object as the new-style header packaging the
+%           old-style header info
+%
+% COMMENT:
+% This should really be static method - it relates to Experiment but does
+% not use Experiment object contents
+%------------------------------------------------------------------------
+
+% Name varargin{1} as (old-style) headers to clarify what is coming in
 headers = varargin{1};
-expdata = repmat(IX_experiment,1,numel(headers));
-instruments = repmat(IX_inst(),1,numel(headers));
-samples = repmat(IX_samp(),1,numel(headers));
 
-% convert old headers, resored differently from sqw and mat files into the
-% same format
+% Make arrays of the hew-style header class types to receive the data
+% coming from each run in headers
+expdata     = repmat(IX_experiment(), 1,numel(headers));
+instruments = repmat(IX_null_inst(),  1,numel(headers));
+samples     = repmat(IX_null_sample(),1,numel(headers));
+
```

Comment:
we never had empty sample in our old Horace. It was always IX_samp (with lattice) . Empty instrument would be ok

Reply:
This is just initializing samples with a set of null samples indicating that the sample is not defined at this point. It will be populated further down this function. IX_samp is now the base class of IX_null_sample and IX_sample and although it is not now abstract to allow serializable to work with it, it should still be treated as abstract in how it is used. Hence IX_null_sample is used here.